### ISSUE TITLE
`reactivity` does not accept `default` or `initial` as names

### ISSUE THREAD

**Issue by N0tExisting** [solidjs-community/eslint-plugin-solid#151]
## **Describe the bug**

When trying to initialize state from props named `default` or `initiall`, an error is shown.

## **To Reproduce**

```tsx
import { createSignal, type FlowProps } from 'solid-js';

export type FailProps = Props<{ default: number, initial: string }>;

export default function Failure(p: FailProps) {
  createSignal(props.default)
  createSignal(props.initial)
}
```

## **Expected behavior**

No Error is shown.

## **Environment (please complete the following information):**

- _**OS:**_ Ubuntu Server 22
- _**Node** version:_ v20.11.0
- _**`eslint-plugin-solid`** version:_ 0.14.2
- _**`eslint`** version:_ 8.57.0

## **Additional context**

I think this might be intentional, but I couldn't find anything to prove that.

- [X] I would be willing to contribute a PR to fix this issue

**Comment by joshwilsonvu** [comment]
https://github.com/solidjs-community/eslint-plugin-solid/blob/main/packages/eslint-plugin-solid/src/rules/reactivity.ts#L578

I wouldn't mind if you changed the regex there to accept `initial` or `default`, that's fine! I don't think we should `static` on its own though, as it really does need to be a prefix to make sense.

**Comment by N0tExisting** [comment]
yeah, it would be useful if [`@once`](https://docs.solidjs.com/reference/jsx-attributes/once) would work on types:
```tsx
export interface MyProps {
  /** @once */
  staticThing: string;
}
```
